src/ImageSharp/ColorSpaces/Conversion/ColorSpaceConverter.Adapt.cs

```diff
@@ -0,0 +1,198 @@
+﻿// <copyright file="ColorSpaceConverter.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.ColorSpaces.Conversion
+{
+    using System;
+
+    using ImageSharp.ColorSpaces;
+    using ImageSharp.ColorSpaces.Conversion.Implementation.Rgb;
+
+    /// <content>
+    /// Performs chromatic adaptation on the various color spaces.
+    /// </content>
+    public partial class ColorSpaceConverter
```

Comment:
Really cool to have all these new features, and the implementation is good and clean for the first iteration! :smile: 
Also good to see it being internal for now, so we don't have to freeze it with the beta! ;)